check/check.go

```diff
@@ -26,6 +26,40 @@ import (
 
 // State is the state of a control check.
 type State string
+type AuditType string
+
+const TypeAudit = "audit"
+
+type Audit struct {
+	audit string
+}
+
+func (a *Audit) Execute(customConfig ...interface{}) (result string, errMessage string, state State) {
```

Comment:
I'd make this a method on `(a Audit)` rather than `(a *Audit)` to avoid having to pass pointers around. (Don't think about performance without doing a [benchmark](https://golang.org/pkg/testing/#hdr-Benchmarks) test as quite often the performance doesn't behave like it does in C)

Reply:
I will happy if u can elaborate little more  ? how we access struct data after yaml.unmarshall  without pointer receiver ?

Reply:
Put that question the other way around - why do you think it needs to be a pointer receiver? You know that you can have methods on any type in Go right? (Well, any type that is defined in the same package).